panic: message "auth.token.parse.error.parseError" not found in language "en"

I entered an invalid offline token value and got a localization error. This comes from the "github.com/redhat-developer/app-services-cli/pkg/auth/token" dependency.

Like you have done with the builder and config, this should alos be copied.

```
❯ export OFFLINE_TOKEN=enda
❯ terraform plan
╷
│ Error: Plugin did not respond
│ 
│   with provider["registry.terraform.io/pmuir/rhoas"],
│   on create-kafka.tf line 10, in provider "rhoas":
│   10: provider "rhoas" {}
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-rhoas_v0.1.0 plugin:

panic: message "auth.token.parse.error.parseError" not found in language "en"

goroutine 67 [running]:
github.com/nicksnyder/go-i18n/v2/i18n.(*Localizer).MustLocalize(0xc00000ca60, 0xc00075f2e0, 0xc00075f300, 0x40cc08)
	github.com/nicksnyder/go-i18n/v2@v2.1.2/i18n/localizer.go:211 +0x79
github.com/redhat-developer/app-services-cli/internal/localizer.MustLocalize(0xc00075f3e0, 0xc0000363ce, 0x4)
	github.com/redhat-developer/app-services-cli@v0.0.0-20210426115030-1f3d8f29f552/internal/localizer/localizer.go:78 +0xc2
github.com/redhat-developer/app-services-cli/internal/localizer.MustLocalizeFromID(...)
	github.com/redhat-developer/app-services-cli@v0.0.0-20210426115030-1f3d8f29f552/internal/localizer/localizer.go:86
github.com/redhat-developer/app-services-cli/pkg/auth/token.Parse(0xc0000363ce, 0x4, 0x203000, 0x203000, 0xc00075f500)
	github.com/redhat-developer/app-services-cli@v0.0.0-20210426115030-1f3d8f29f552/pkg/auth/token/token.go:85 +0x123
github.com/redhat-developer/app-services-cli/pkg/auth/token.GetExpiry(0xc0000363ce, 0x4, 0xc01a65773a6ece76, 0x66b9c18, 0x1e2f5a0, 0x1094540, 0xc00075f530, 0x40cc08, 0x30)
	github.com/redhat-developer/app-services-cli@v0.0.0-20210426115030-1f3d8f29f552/pkg/auth/token/token.go:109 +0x4d
github.com/redhat-developer/app-services-cli/pkg/auth/token.(*Token).IsValid(0xc0005ebf20, 0xc0005ebf50, 0x98, 0xc00075f5d0)
	github.com/redhat-developer/app-services-cli@v0.0.0-20210426115030-1f3d8f29f552/pkg/auth/token/token.go:37 +0x97
redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/connection.(*Builder).BuildContext(0xc00075f898, 0x1569b00, 0xc000138210, 0x1094540, 0xc0002e6480, 0x1e16401)
	redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/connection/builder.go:191 +0x224
redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/connection.(*Builder).Build(...)
	redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/connection/builder.go:143
redhat.com/rhoas/rhoas-terraform-provider/m/rhoas.providerConfigure(0x1569b80, 0xc0005ebc50, 0xc00038e380, 0x0, 0xc0005ebbc0, 0x0, 0x0, 0x0)
	redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/provider.go:82 +0x26f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0xc00030e780, 0x1569b80, 0xc0005ebc50, 0xc0005ebbc0, 0x11a7320, 0xc00004c240, 0x1569b80)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.2/helper/schema/provider.go:286 +0x254
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0xc00072b300, 0x1569ac0, 0xc00004c180, 0xc00012dfc0, 0xc00004c180, 0x11f6660, 0x124cfc0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.2/helper/schema/grpc_provider.go:523 +0x3b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).Configure(0xc00012c5a0, 0x1569ac0, 0xc00004c180, 0xc00004c140, 0xc00012c5a0, 0xc0005eb680, 0xc0002a0ba0)
	github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/server/server.go:182 +0xca
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler(0x124cfc0, 0xc00012c5a0, 0x1569b80, 0xc0005eb680, 0xc000306f00, 0x0, 0x1569b80, 0xc0005eb680, 0xc000330700, 0x3a)

```

--- a/.gitignore
+++ b/.gitignore
@@ -68,3 +68,4 @@ terraform
 
 # IDEs
 .idea/*
+terraform-provider-rhoas

--- a/go.mod
+++ b/go.mod
@@ -2,7 +2,7 @@ module redhat.com/rhoas/rhoas-terraform-provider/m
 
 require (
 	github.com/Nerzal/gocloak/v7 v7.11.0
-	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
+	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/hashicorp/terraform-plugin-docs v0.4.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.4.2
 	github.com/pkg/errors v0.9.1

--- a/go.mod
+++ b/go.mod
@@ -2,6 +2,7 @@ module redhat.com/rhoas/rhoas-terraform-provider/m
 
 require (
 	github.com/Nerzal/gocloak/v7 v7.11.0
+	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
 	github.com/hashicorp/terraform-plugin-docs v0.4.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.4.2
 	github.com/pkg/errors v0.9.1

--- a/rhoas/cli/auth/token/token.go
+++ b/rhoas/cli/auth/token/token.go
@@ -1,0 +1,128 @@
+package token
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/redhat-developer/app-services-cli/pkg/logging"
+)
+
+// Token contains the current access and refresh tokens from the Authorization server
+type Token struct {
+	Logger       logging.Logger
+	AccessToken  string `json:"access_token,omitempty" doc:"Bearer access token."`
+	RefreshToken string `json:"refresh_token,omitempty" doc:"Offline or refresh token."`
+}
+
+func (t *Token) IsValid() (tokenIsValid bool, err error) {
+	now := time.Now()
+	if t.AccessToken != "" {
+		var expires bool
+		var left time.Duration
+		expires, left, err = GetExpiry(t.AccessToken, now)
+		if err != nil {
+			return
+		}
+		if !expires || left > 5*time.Second {
+			tokenIsValid = true
+			return
+		}
+	}
+	if t.RefreshToken != "" {
+		var expires bool
+		var left time.Duration
+		expires, left, err = GetExpiry(t.RefreshToken, now)
+		if err != nil {
+			return
+		}
+		if !expires || left > 10*time.Second {
+			tokenIsValid = true
+			return
+		}
+	}
+	return
+}
+
+// NeedsRefresh checks if the access token is missing,
+// expired or nearing expiry and should be refreshed
+func (t *Token) NeedsRefresh() bool {
+	if t.AccessToken == "" && t.RefreshToken != "" {
+		return true
+	}
+
+	now := time.Now()
+	expires, left, err := GetExpiry(t.AccessToken, now)
+	if err != nil {
+		t.Logger.Debug(fmt.Sprintf("token expired: %s", err.Error()))
+		return false
+	}
+
+	if !expires || left > 5*time.Minute {
+		return false
+	}
+
+	return true
+}
+
+func Parse(textToken string) (token *jwt.Token, err error) {
+	parser := new(jwt.Parser)
+	token, _, err = parser.ParseUnverified(textToken, jwt.MapClaims{})
+	if err != nil {
+		err = fmt.Errorf("parse error: %v", err)
+		return
+	}
+	return token, nil
+}
+
+func MapClaims(token *jwt.Token) (jwt.MapClaims, error) {
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		err := fmt.Errorf("error in claims %v", claims)
+		return nil, err
+	}
+
+	return claims, nil
+}
+
+func GetExpiry(tokenStr string, now time.Time) (expires bool,
+	left time.Duration, err error) {
+
+	token, err := Parse(tokenStr)
+	if err != nil {
+		return false, 0, err
+	}
+
+	claims, err := MapClaims(token)
+	if err != nil {
+		return false, 0, err
+	}
+	var exp float64
+	claim, ok := claims["exp"]
+	if ok {
+		exp, ok = claim.(float64)
+		if !ok {
+			return false,time.Second, fmt.Errorf("expeted expire claim but got %v", claim)
+		}
+	}
+	if exp == 0 {
+		expires = false
+		left = 0
+	} else {
+		expires = true
+		left = time.Unix(int64(exp), 0).Sub(now)
+	}
+
+	return
+}
+
+func GetUsername(tokenStr string) (username string, ok bool) {
+	accessTkn, _ := Parse(tokenStr)
+	tknClaims, _ := MapClaims(accessTkn)
+	userName, ok := tknClaims["preferred_username"]
+	if ok {
+		username = fmt.Sprintf("%v", userName)
+	}
+
+	return username, ok
+}

--- a/rhoas/cli/auth/token/token.go
+++ b/rhoas/cli/auth/token/token.go
@@ -69,7 +69,7 @@ func Parse(textToken string) (token *jwt.Token, err error) {
 	parser := new(jwt.Parser)
 	token, _, err = parser.ParseUnverified(textToken, jwt.MapClaims{})
 	if err != nil {
-		err = fmt.Errorf("parse error: %v", err)
+		err = fmt.Errorf("parse error: %w", err)
 		return
 	}
 	return token, nil
@@ -102,7 +102,7 @@ func GetExpiry(tokenStr string, now time.Time) (expires bool,
 	if ok {
 		exp, ok = claim.(float64)
 		if !ok {
-			return false,time.Second, fmt.Errorf("expeted expire claim but got %v", claim)
+			return false, time.Second, fmt.Errorf("expeted expire claim but got %v", claim)
 		}
 	}
 	if exp == 0 {

--- a/rhoas/cli/connection/builder.go
+++ b/rhoas/cli/connection/builder.go
@@ -7,14 +7,13 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/auth/token"
 	"redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/build"
 	"redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/config"
 
 	"github.com/redhat-developer/app-services-cli/pkg/cmd/debug"
 
 	"github.com/Nerzal/gocloak/v7"
-
-	"github.com/redhat-developer/app-services-cli/pkg/auth/token"
 	"github.com/redhat-developer/app-services-cli/pkg/logging"
 )
 

--- a/rhoas/cli/connection/keycloak_connection.go
+++ b/rhoas/cli/connection/keycloak_connection.go
@@ -22,7 +22,6 @@ import (
 	"github.com/redhat-developer/app-services-cli/pkg/logging"
 
 	"github.com/Nerzal/gocloak/v7"
-
 )
 
 var DefaultScopes = []string{

--- a/rhoas/cli/connection/keycloak_connection.go
+++ b/rhoas/cli/connection/keycloak_connection.go
@@ -7,6 +7,7 @@ import (
 	"net"
 	"net/http"
 	"net/url"
+	"redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/auth/token"
 	"redhat.com/rhoas/rhoas-terraform-provider/m/rhoas/cli/config"
 
 	"github.com/redhat-developer/app-services-cli/pkg/api/ams/amsclient"
@@ -22,7 +23,6 @@ import (
 
 	"github.com/Nerzal/gocloak/v7"
 
-	"github.com/redhat-developer/app-services-cli/pkg/auth/token"
 )
 
 var DefaultScopes = []string{

